kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/group/GroupMetadataManager.java

```diff
@@ -582,6 +583,9 @@ public GroupMetadata addGroup(GroupMetadata group) {
                                 + " when appending to log due to ",
                         filteredOffsetMetadata, group.groupId(), consumerId, group.generationId(), cause);
 
+                if (cause.getCause() instanceof PulsarClientException.AlreadyClosedException) {
```

Comment:
I think we can try to remove the producer from the cache.

```java
                    offsetsProducers.remove(partitionFor(group.groupId()))
```

After that, there will be no need to call `handler.getGroupCoordinator().getOffsetsProducers().clear()` in the test.

Reply:
+1 good catch

Reply:
Good catch. Addressed.